TFT/src/User/API/Vfs/vfs.c

```diff
@@ -272,19 +272,16 @@ bool getPrintTitle(char * buf, uint8_t len)
 {
   // example: "SD:/test/cap2.gcode" -> "SD:cap2.gcode"
 
-  char * strPtr = getPrintFilename();
-
-  // "+ 2": space for terminating null character and the flag for filename extension check
-  if (strlen(getFS()) + strlen(strPtr) + 2 > len)
+  // '\0': terminating null character
+  //  "+ 1": space for flag for filename extension check
+  if (snprintf(buf, len, "%s%s%c", getFS(), getPrintFilename(), '\0') + 1 > len)
```

Comment:
current code = safe
proposed code = unsafe

Reply:
In a very-very-very unrealistic specific scenario the extension wouldn't be hidden:
- most of the characters in the filename being unicode (more than 1 byte)
- the filename before extension less than (LCD_WIDTH - BYTE_WIDTH) characters (with the above characteristics)
- total number of bytes exceeds MAX_TITLE_LEN and there's a partial extension (ex.: .gcod)

I wouldn't call it "unsafe" but everyone is entitled to an opinion...
Anyway, even that scenario is covered now.

Reply:
Outdated